logstash-core/src/main/java/org/logstash/common/BufferedTokenizerExt.java

```diff
@@ -55,7 +55,11 @@ public IRubyObject init(final ThreadContext context, IRubyObject[] args) {
             this.delimiter = args[0].convertToString();
         }
         if (args.length == 2) {
-            this.sizeLimit = args[1].convertToInteger().getIntValue();
+            final int sizeLimit = args[1].convertToInteger().getIntValue();
+            if (sizeLimit <= 0) {
```

Comment:
We had talks/ideas to ignore setting the limit if specific value (for example negative -1) or zero. It has an interface accepts a single delimiter to ignore limit setting. So, I would like _only_ to make this limit validated and not add any other hidden features.

Reply:
I agree.

As it stands, this PR only prevents BufTok from being initialized with values that could _never_ result in bytes being emitted :+1: